dln/operator.py

```diff
@@ -361,21 +378,66 @@ def instantiate_tokenizer(model_name: str):
         encoder = AutoTokenizer.from_pretrained(pretrained_path)
     return encoder
 
-class LLMRegistry:
-    def __init__(self):
-        self.models = []
 
-    def register(self, model_name: str, **generation_options) -> LLM:
-        if model_name in GPT.AVAILABLE_MODELS:
-            llm = GPT(model_name, **generation_options)
-        else:
-            llm = VLLM(model_name, **generation_options)
+class LLMRegistry:
 
-        self.models.append(llm)
+    def __init__(self, config=None):
+        self.models : Dict[str, LLM] = {}
+        if config is not None:
+            self._load_from_configs(config)
+
+    def register(self, model_name: str, model_type: str = None, **generation_options) -> LLM:
+        """ Register a single model to the registry.
+        Args:
+            model_name: how you refer to the model, for example: gpt-3.
+            model_type: the api model name, for example: text-davinci-003. If not provided, use model_name as default.
+            **generation_options: generation options, for example: api_key, api_base, api_type, api_version, max_tokens, temperature, etc.
+        Returns:
+            the instantiated model
+        """
+        if model_type is None:
```

Comment:
What happens if that model name was already registered?

Reply:
At the moment it overrides the previous model, but we can raise an exception instead. Which one makes more sense for you?

Reply:
It should have a unique UUID. Do you see any issues with having multiple instances of a same model?

Reply:
I don't see any issue with having multiple instances of the same model. We register the same model using different default generation options (temperature) for backward and forward models, for example.

Reply:
I would prefer the user to specify a name for every llm registered, and the register to raise an exception if the `model_name` already exists. The use case I have in mind (from llama2_scripts branch, and that is partially implemented here) is to register many models from a config file and refer to them by name everywhere else. What do you think?

Reply:
That makes sense. I agree it is better to let the user have entire control over the model name.

Reply:
Sounds good. Register now raises an exception if model name already exists.